runtime/Java/src/org/antlr/v4/runtime/atn/ATNDataWriter.java

```diff
@@ -0,0 +1,106 @@
+package org.antlr.v4.runtime.atn;
+
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+public class ATNDataWriter {
```

Comment:
If we go base64 route, shouldn't we use existing encoder/decoders to avoid bugs? https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html

Reply:
Sure we can. But I tried to get rid of excess big array allocations in the deserializer (there are 2 additional allocations if use standard methods). Since it may affect application startup time.

Reply:
well, if it's a trade between an alloc that gets collected vs homegrown base64, it seems safer to take the extra GC hit.